fix(test): WASM tests fail when WASM binary is not built

## Problem

Running `pnpm test` without a prior WASM build causes 18 test failures in `__test__/wasm.spec.ts`. This is the default path for new contributors following the setup guide.

**Reproduction steps:**

```bash
git clone https://github.com/derodero24/rapid-fuzzy.git
cd rapid-fuzzy
pnpm install --ignore-scripts
pnpm run build          # builds native .node only
pnpm test               # 18 WASM tests fail
```

**Error pattern:**

```
AssertionError: expected 'undefined' to be 'function'
TypeError: wasm.<fn> is not a function
```

All 18 failures are in `wasm.spec.ts` because the `.wasm` binary does not exist (only the native `.node` was built).

## Expected behavior

Running `pnpm test` after the standard setup in `CONTRIBUTING.md` should not produce unexpected failures.

## Current contributor experience

- `CONTRIBUTING.md` documents `pnpm run build` + `pnpm test` as the setup flow
- `pnpm run build` only builds the native Node.js addon (requires no WASM toolchain)
- WASM build (`pnpm run build:wasm`) requires the `wasm32-wasip1-threads` Rust target, which is not mentioned in `CONTRIBUTING.md`

## Proposed solutions

**Option A — Skip WASM tests when binary is absent (recommended):**
Add a check at the top of `wasm.spec.ts` that skips the entire suite if the WASM module is not available:

```typescript
import { describe } from 'vitest';

const wasmAvailable = /* check if wasm module loads */;

describe.skipIf(!wasmAvailable)('WASM', () => {
  // existing tests
});
```

This way `pnpm test` always passes in the standard setup, and WASM tests run automatically when the binary exists.

**Option B — Document WASM build in CONTRIBUTING.md:**
Add the WASM build step as optional in the setup guide, and note that WASM tests will be skipped/fail without it.

**Option C — Both A and B:**
Skip tests gracefully AND document the WASM build step for contributors who want to run the full suite.

## Context

- WASM tests pass in CI because the CI build matrix includes `wasm32-wasip1-threads`
- The 282 non-WASM tests all pass without issue
- Node.js emits `ExperimentalWarning: WASI is an experimental feature` during WASM tests (informational, not blocking)

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -27,6 +27,21 @@ pnpm test          # JS/TS tests
 cargo test         # Rust unit tests
 ```
 
+### WASM testing (optional)
+
+WASM tests are automatically skipped if the WASM binary is not built. To run the full test suite including WASM:
+
+```bash
+# Install the WASM target
+rustup target add wasm32-wasip1-threads
+
+# Build the WASM binary
+pnpm run build:wasm
+
+# Run all tests including WASM
+pnpm test
+```
+
 ### GitHub Codespaces / Dev Containers
 
 You can skip local setup entirely by using [GitHub Codespaces](https://github.com/codespaces) or [VS Code Dev Containers](https://code.visualstudio.com/docs/devcontainers/containers). The repository includes a devcontainer configuration with all required tools pre-installed.

--- a/__test__/wasm.spec.ts
+++ b/__test__/wasm.spec.ts
@@ -1,16 +1,22 @@
-import { existsSync } from 'node:fs';
-import { resolve } from 'node:path';
 import { describe, expect, it } from 'vitest';
 
-const wasmBinaryPath = resolve(__dirname, '../rapid-fuzzy.wasm32-wasi.wasm');
-const wasmExists = existsSync(wasmBinaryPath);
-
-// Skip all WASM tests if the binary is not built
-// Build with: pnpm run build --target wasm32-wasip1-threads
-describe.skipIf(!wasmExists)('wasm', () => {
+// Check if WASM module is available and functional.
+// Handles both missing binary (not built) and stale binary (outdated build).
+let wasmAvailable = false;
+// biome-ignore lint/suspicious/noExplicitAny: WASM module has dynamic exports
+let wasm: Record<string, any> = {};
+try {
   // eslint-disable-next-line @typescript-eslint/no-require-imports
-  const wasm = wasmExists ? require('../rapid-fuzzy.wasi.cjs') : {};
+  const mod = require('../rapid-fuzzy.wasi.cjs');
+  wasmAvailable = typeof mod.levenshtein === 'function';
+  if (wasmAvailable) wasm = mod;
+} catch {
+  wasmAvailable = false;
+}
 
+// Skip all WASM tests if the module is not available or outdated
+// Build with: pnpm run build:wasm
+describe.skipIf(!wasmAvailable)('wasm', () => {
   describe('exports', () => {
     it('should export all expected functions', () => {
       const expectedExports = [

